src/Pages/Profile/Profile.tsx

```diff
@@ -25,6 +25,14 @@ class ProfilePage extends Component<RouteComponentProps<any>, IStates> {
     }
   }
 
+  public componentWillReceiveProps(nextProps: any) {
```

Comment:
With the new code after React 16.3, you should named this function is `UNSAFE_componentWillReceiveProps` instead. Because `componentWillReceiveProps` will continue to work until React version 17.
Or you can use `getDerivedStateFromProps`. 
https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops